### ISSUE TITLE
RFE: Be able to disable following ldap referrals

### ISSUE THREAD

**Issue by opoplawski** [opensearch-project/security#195]
Like the elasticsearch xpack plugin (https://www.elastic.co/guide/en/elasticsearch/reference/current/security-settings.html) it would be good to be able to disable following ldap referrals.  It can generate a lot of failed connections when connecting to AD LDAP servers.

**Comment by alolita** [comment]
Thanks! @opoplawski

**Comment by dignajar** [comment]
I would like to have this parameter as well to avoid warning such as:

```
[2020-03-19T13:36:59,401][WARN ][o.l.r.SearchReferralHandler] [es-client-6964bdbd4b-gfdxj] Could not follow referral to ldaps://ForestDnsZones.<domain>/DC=ForestDnsZones,DC=xx,DC=xx
```

```
[2020-03-19T13:36:59,384][WARN ][o.l.r.SearchReferralHandler] [es-client-6964bdbd4b-gfdxj] Could not follow referral to ldaps://DomainDnsZones.<domani>/DC=DomainDnsZones,DC=xx,DC=xx
```

Thanks!

**Comment by philipp1992** [comment]
we would also need this

**Comment by daskanu** [comment]
Yes, please implement this. We also need this feature.

Thanks in advance.

**Comment by ohdearaugustin** [comment]
Same here.

**Comment by voyagersm** [comment]
Also need this.

**Comment by konvergence** [comment]
Hi,

By default the follow_referrals is set on true.
But if there is any  firewall rules that deny the connexion on th other DC domain, we get a partial authentication into opendistro and we can not get provisionned roles on the external identity.

Could you allow to configure   the option follow referrals ?
Or continue the authorization process even if an exception is trigger on follow referrals ?

Regards

**Comment by fabide** [comment]
we would need this as well

**Comment by Brazier85** [comment]
+1

**Comment by krylov-aleksandr** [comment]
Same here.

**Comment by mfvanek** [comment]
+1

**Comment by ChBerrich** [comment]
+1

**Comment by justpolidor** [comment]
Hello, any news about this issue?

**Comment by davidlago** [comment]
Thanks for the reminder @justpolidor. I've marked this as an enhancement instead of a bug (since I believe that better captures the type of issue) and added the "good first issue" label to it as this could be a good one in case someone is looking for that first PR to tackle and get this one off the list!

**Comment by aruzhannurman** [comment]
Also need this.

**Comment by peternied** [comment]
@aruzhannurman This feature should be available in the next release of OpenSearch, v2.7, track it here
- https://github.com/opensearch-project/opensearch-build/issues/3230
